service/storage_service.cc

```diff
@@ -6690,7 +6690,9 @@ future<> storage_service::move_tablet(table_id table, dht::token token, locator:
             throw std::runtime_error(format("Unknown host: {}", dst.host));
         }
         if (dst.shard >= node->get_shard_count()) {
-            throw std::runtime_error(format("Host {} does not have shard {}", dst.shard));
+            throw std::runtime_error(format("Host {} does not have shard {}",
+                                            get_token_metadata().get_my_id(),
```

Comment:
Actually I think this is about the remote node, so you need to print node->id() here.

Reply:
thank you Botond! indeed. fixed and repushed.